content/dotnet-template-azure-iot-edge-module/CSharp/ModuleBackgroundService.cs

```diff
@@ -0,0 +1,50 @@
+﻿using Microsoft.Azure.Devices.Client;
+using Microsoft.Azure.Devices.Client.Transport.Mqtt;
+using System.Text;
+
+namespace SampleModule;
+
+internal class ModuleBackgroundService : BackgroundService
+{
+    private int _counter;
+    private ModuleClient? _moduleClient;
+    private readonly ILogger<ModuleBackgroundService> _logger;
+
+    public ModuleBackgroundService(ILogger<ModuleBackgroundService> logger) => _logger = logger;
+
+    protected override async Task ExecuteAsync(CancellationToken stoppingToken)
```

Comment:
Shall we save off the cancellation token as a member variable and use it in `PipeMessage` so it can shutdown gracefully too?

Reply:
I dont see a need for gracefully shutdown

Reply:
But it could help it shut down faster if the container waits on a process to exit.

I mean, either use cancellation tokens or don't. I don't see a point in only using them sometimes.

Reply:
got it, updated